pkg/apis/networking/v1alpha1/ingress_validation.go

```diff
@@ -166,3 +167,12 @@ func (t *IngressTLS) Validate(ctx context.Context) *apis.FieldError {
 	}
 	return all
 }
+
+// Validate inspects and validates HTTPOption object.
+func (t HTTPOption) Validate(ctx context.Context) *apis.FieldError {
+	var all *apis.FieldError
+	if t != HTTPOptionEnabled && t != HTTPOptionDisabled && t != HTTPOptionRedirected {
```

Comment:
it is more customary to use `switch` here.

Reply:
Not sure I understand how. Can you pls suggest a change?